### ISSUE TITLE
protodoc describing map fields incorrectly

### ISSUE THREAD

**Issue by ravenblackx** [envoyproxy/envoy#32523]
Example error, on https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto

> **typed_per_filter_config**
(**repeated** map<[string](https://developers.google.com/protocol-buffers/docs/proto#scalar), [Any](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#any)>) This field can be used to provide virtual host level per filter config. The key should match the [filter config name](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpfilter-name). See [Http filter route specific config](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_filters#arch-overview-http-filters-per-filter-config) for details.

In the proto file [we can see](https://github.com/envoyproxy/envoy/blob/ffbf1889ef788ccb4b1d0e402ed2c3976d7204fd/api/envoy/config/route/v3/route.proto#L150-L150) that it is in fact just a map field. I'm not sure there is even such a thing as a repeated map field.

**Comment by phlax** [comment]
> I'm not sure there is even such a thing as a repeated map field.

is this not just a map field which _is_ repeated?

**Comment by phlax** [comment]
(not sure what this should be however)

**Comment by lizan** [comment]
This might be a bug of the proto doc generation, map is internally represented as repeated pair<K, V> so itself couldn't be repeated.

**Comment by ravenblackx** [comment]
Yeah, it would be accurate to describe it as a `map<string, Any>`, or a `repeated pair<string, Any>`, but not a `repeated map<>`.

**Comment by phlax** [comment]
not sure exactly the problem/solution but the code in question is here https://github.com/envoyproxy/envoy/blob/790abeba40a62cd0bfecad43862ef79e12ce421a/tools/protoprint/protoprint.py#L415-L441
